### DISCUSSION TITLE
How can I access the page data when a custom tag (shortcode) is rendered?

### DISCUSSION THREAD

**Discussion by gabipetrovay** [11ty/eleventy#1591] [Q&A]
Having such a page with the YAML front-matter:

```yml
---
title: this is front-matter data
---
Page content
{% myCustomTag valueOfArg1 %}
```

How can I access the front-matter page data in the handler function of the `customTag` shortcode?

```javascript
eleventyConfig.addShortcode('myCustomTag', function(arg1) {
    // how can I access here the page front-matter data? E.g. "title"
    return `<div>...${ arg1 }...</div>`;
});
```

Posted also on [Stack Overflow](https://stackoverflow.com/q/65762089/454103).

**Comment by pdehaan** [comment]
Possibly related https://github.com/11ty/eleventy/issues/741

**Reply by gabipetrovay** [reply]
#741 talks about the `page` data and not the page front-matter (referred in #741 as context or as data cascade). #741 also does not answer the latest questions (after closing) on this matter: can one access the page front-matter/context/data cascade from the shortcode code? If not, are there plans to have it available in the future? Also #1044 makes no statements on this.

**Comment by solution-loisir** [comment]
Hi @gabipetrovay, I guess that, in your case, you can pass `title` as an argument.
```twig
---
title: Your Title
---
{% myCustomTag title %}
```

**Reply by gabipetrovay** [reply]
I want to use custom tags that **collect data from the page front matter** rather than having the page editor supply it every time. This simplifies editing of page sources AND minimises human errors/page build failures:

```
---
myCustomTagDefaultArg: my value
---
{% myCustomTag %}
...
{% myCustomTag %}
...
{% myCustomTag 'this is an override' %}
...
{% myCustomTag %}
```

**Comment by mahnunchik** [comment]
I think page data may be passed to filters, shortcodes, and transforms in the way `eleventyComputed` already does. https://github.com/11ty/eleventy/issues/2135
